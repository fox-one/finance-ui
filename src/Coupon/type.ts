export interface Coupon {
  type: 'DISCOUNT' | 'VOUCHER';
  amount: number;
  name: string;
  expired_at: string | number;
  [props: string]: any;
}