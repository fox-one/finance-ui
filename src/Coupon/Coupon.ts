import {
  Component,
  Vue,
  Prop
} from 'vue-property-decorator';
import classnames from '@utils/classnames';
/* import types */
import type { CreateElement, VNode } from 'vue';


export interface Coupon {
  type: 'DISCOUNT' | 'VOUCHER';
  amount: number;
  name: string;
  expired_at: string | number;
  [props: string]: any;
}

@Component
export class Coupon extends Vue {
  @Prop({ type: Array, default: () => [] }) private coupons!: Coupon[];
  @Prop({ type: String, default: '' }) private className!: string;

  public render(h: CreateElement): VNode {
    const content = this.$slots.default;
    const classes = classnames('coupon');
    return h(
      'div',
      {
        staticClass: classes(),
        class: this.className
      },
      content
    );
  }
}

export default Coupon;