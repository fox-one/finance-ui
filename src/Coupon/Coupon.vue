<template>
  <div :class="classes(void 0, className)">
    <slot name="default" />
  </div>
</template>

<script lang="ts">
import {
  defineComponent,
  PropType,
  onMounted
} from '@vue/composition-api';
import classnames from '@utils/classnames';
// import type { Coupon } from './type';

export default defineComponent({
  name: 'Coupon',
  props: {
    className: {
      type: String,
      default: ''
    },
    coupons: {
      type: Array,
      default: () => []
    }
  },
  setup() {
    const classes = classnames('coupon');
    onMounted(() => {
      console.info('Coupon mounted!');
    });

    return { classes };
  }
});
</script>