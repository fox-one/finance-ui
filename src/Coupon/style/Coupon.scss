@import '../../styles/variable.scss';

.#{$prefix}-coupon {
  display: block;
  color: green;
}
