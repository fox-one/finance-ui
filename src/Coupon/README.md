# Coupon

## Example

```tsx
import { Coupon } from './Coupon';

<coupon />
```
