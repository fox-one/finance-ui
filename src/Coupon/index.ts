import Coupon from './Coupon';

export { Coupon } from './Coupon';

export default Coupon;
