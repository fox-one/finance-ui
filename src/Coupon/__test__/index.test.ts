import { shallowMount, mount } from '@vue/test-utils';
import Coupon from '../';

describe('Coupon', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(Coupon);
    expect(wrapper).toMatchSnapshot();
  });
});
