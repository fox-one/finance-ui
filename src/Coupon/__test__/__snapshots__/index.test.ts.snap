// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coupon renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {
    "hook:beforeMount": Array [
      Array [],
    ],
    "hook:created": Array [
      Array [],
    ],
    "hook:mounted": Array [
      Array [],
    ],
  },
  "_emittedByOrder": Array [
    Object {
      "args": Array [],
      "name": "hook:created",
    },
    Object {
      "args": Array [],
      "name": "hook:beforeMount",
    },
    Object {
      "args": Array [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;
